### DISCUSSION TITLE
[SUGGESTION] Auto generated foreign key and index names can be too long, should detect this

### DISCUSSION THREAD

**Discussion by deanlj** [TiagoSilvaPereira/vemto-issues#145] [Ideas & Suggestions]
When generating Foreign keys and indexes there isn't a check for the key / index name being too long.

Ideally this should be detected and ask/require a unique foreign key or index name

To reproduce

- Create a model called WorkflowCondition
- Create a model called WorkflowStepTask
- Create a belongs to many relationship between them
- run migrations
<img width="1440" alt="Screen Shot 2021-04-30 at 20 46 16" src="https://user-images.githubusercontent.com/4245429/116746963-29e6b800-a9f5-11eb-977e-18799fc7c013.png">

Migrating: 2021_04_22_009009_add_foreigns_to_workflow_conditions_workflow_step_tasks_table

   Illuminate\Database\QueryException

  SQLSTATE[42000]: Syntax error or access violation: 1059 Identifier name 'workflow_conditions_workflow_step_tasks_workflow_step_task_id_foreign' is too long (SQL: alter table `workflow_conditions_workflow_step_tasks` add constraint `workflow_conditions_workflow_step_tasks_workflow_step_task_id_foreign` foreign key (`workflow_step_task_id`) references `workflow_step_tasks` (`id`))

  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:678
    674▕         // If an exception occurs when attempting to run a query, we'll format the error
    675▕         // message to include the bindings with SQL, which will make this exception a
    676▕         // lot more helpful to the developer instead of just the database's errors.
    677▕         catch (Exception $e) {
  ➜ 678▕             throw new QueryException(
    679▕                 $query, $this->prepareBindings($bindings), $e
    680▕             );
    681▕         }
    682▕

      +9 vendor frames
  10  database/migrations/2021_04_22_009009_add_foreigns_to_workflow_conditions_workflow_step_tasks_table.php:28
      Illuminate\Support\Facades\Facade::__callStatic("table")

      +21 vendor frames
  32  artisan:37
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))

**Expected behavior**
Specify a unique foreign key or index name

**Screenshots**
Attached

**Desktop (please complete the following information):**
 - Mac Os 10.4
 - Browser N/A
 - Version N/A

**Comment by TiagoSilvaPereira** [comment]
Hi @deanlj as it is more a database issue than a Vemto bug, I'll change it for a SUGGESTION issue (currently, you can fix it by passing the index name on your migration: https://stackoverflow.com/questions/30170268/laravel-5-migration-identifier-name-too-long). Thanks

**Comment by deanlj** [comment]
Ok appreciate that, however it does require manual editing of migrations after the app has been generated.

so if the issue isn't addressed it means after the app is generated i have to 
run artisan migrate
see which migration fails
manually edit it to have a short foreign key name
rerun artisan migrate and repeat

currently have 12 migrations failing in one project

maybe it's a mysql issue but would be good if vemto was aware of it
